### PR TITLE
📄Add line/column info to markdown errors

### PR DESCRIPTION
**Pull request by yufeih**
Implements https://ceapex.visualstudio.com/Engineering/_workitems/edit/72242/
Depending on v2 change #4127 
Not all errors has line info, this PR only adds line info errors raised as part of markdown parsing.

#2874

### CODE REVIEW

File: docs/specs/resolve.yml [hunk 1]
```diff
@@ -237,7 +237,7 @@ outputs:
   docs/a.json: |
     {"content":"<p>Link to <a href=\"../dep/logo.svg\">dep</a></p>"}
   build.log: |
-    ["warning","link-is-dependency","File 'logo.svg' referenced by link '../dep/logo.svg' will not be built because it is from a dependency docset","docs/a.md"]
+    ["warning","link-is-dependency","File 'logo.svg' referenced by link '../dep/logo.svg' will not be built because it is from a dependency docset","docs/a.md","",1,9]
```
Comment (OsmondJiang) [inline docs/specs/resolve.yml]: why is there empty string for most of the error message?

Comment (yufeih) [reply docs/specs/resolve.yml]: It is `jsonpath`, we could consider turning it into an object

Comment (OsmondJiang) [reply docs/specs/resolve.yml]: let's add a comment or todo :)

Comment (yufeih) [reply docs/specs/resolve.yml]: https://github.com/dotnet/docfx/issues/4141

**Pull request merged**